### PR TITLE
Initial working draft for Firebase support

### PR DESCRIPTION
**Pull request by nellessen**
An initial working draft for Firebase support. That is included:
- README 
  - Setup Firebase
  - Setup project
  - Running locally
  - Deployment to Firebase
- A simple Firebase Cloud Function (v1) that rights incoming data into BigQuery
- The BigQuery Table is created on the first API request

The table schema and the actual function logic is boilerplate for now and needs to be filled with Elbwalker specific logic by @alexanderkirtzel.

### CODE REVIEW

File: src/firebase/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+### 5. Run locally
+To build and run the code locally, just run from the folder `src/firebase/functions`:
+```
+npm run serve -- --project YOUR_FIREBASE_PROJECT_ID
+``` 
+
+In the console you will find the enpoint URL, e.g. `http://127.0.0.1:5001/syte-elbwalker-test/europe-west1/handleEvent`. You can then send an event to your local API with:
+```
+curl -d '{"key1":"value1", "key2":"value2"}' -H "Content-Type: application/json" -X POST http://127.0.0.1:5001/syte-elbwalker-test/europe-west1/handleEvent
```
Comment (quassy) [inline src/firebase/README.md]: For me it's in the europe-west3 region.

```
curl -d '{"key1":"value1", "key2":"value2"}' -H "Content-Type: application/json" -X POST http://127.0.0.1:5001/syte-elbwalker-test/europe-west1/handleEvent
Function europe-west1-handleEvent does not exist, valid functions are: europe-west3-handleEvent
```

File: src/firebase/functions/src/bigquery.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  {
+    name: "body",
+    type: "STRING",
+    description: "The raw request body as a string",
+  },
+  {
+    name: "headers",
+    type: "STRING",
+    description: "The raw request body as a string",
+  },
```
Comment (quassy) [inline src/firebase/functions/src/bigquery.ts]: Should the be nullable (as they are right now) or required?

File: src/firebase/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+
+### 3. Set up your environment and the Firebase CLI
+Assuming you have `nvm` (Node Version Manager) installed, run:
+```
+nvm install 18
+nvm use 18
+npm install -g firebase-tools
+npm install firebase-functions@latest firebase-admin@latest --save
+npm install @google-cloud/bigquery
```
Comment (quassy) [inline src/firebase/README.md]: Let's get rid of this (we have `npm install` inside functions and also our dependencies there), package.json and package-lock.json here and run everything from inside functions directory.

File: src/firebase/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+For details see the instructions on https://firebase.google.com/docs/functions/get-started?gen=2nd#set-up-your-environment-and-the-firebase-cli for Node.js
+
+
+### 4. Initialize your project
+Firt run `firebase login` to log in via the browser and authenticate the Firebase CLI.
```
Comment (quassy) [inline src/firebase/README.md]: I think this should come after `npm install -g firebase-tools`

File: src/firebase/README.md (hunk 1 above)
Comment (nellessen) [reply src/firebase/README.md]: It's just an example

File: src/firebase/README.md (hunk 4 above)
Comment (nellessen) [reply src/firebase/README.md]: It comes after `npm install -g firebase-tools`. Not sure what you mean

File: src/firebase/functions/src/bigquery.ts (hunk 2 above)
Comment (nellessen) [reply src/firebase/functions/src/bigquery.ts]: @alexanderkirtzel should later define the proper schema!

**Pull request merged**